MmSupervisorPkg/Library/SecurePolicyLib/MemPolicy.c

```diff
@@ -49,6 +51,22 @@ typedef union {
 
 #pragma pack()
 
+/**
+  This function check if the buffer is fully inside MMRAM.
+
+  @param Buffer  The buffer start address to be checked.
+  @param Length  The buffer length to be checked.
+
+  @retval TRUE  This buffer is not part of MMRAM.
+  @retval FALSE This buffer is from MMRAM.
+**/
+BOOLEAN
+EFIAPI
+IsBufferInsideMmram (

```

Comment:
This is a catch: the module that uses this needs to have its own implementation. As per scope of this initiative, the supv core has an implementation based on global variables, whereas the spam core will always query the MSR to get the value. This is because the same routine will run on all cores, so saving to global for spam core is not as safe. We run its validation routine once anyway.

Thoughts?

Reply:
Seems to be fine for now.